src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,224 @@
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        userDao.save(user1);
+        userDao.save(user2);
+        userDao.save(user3);
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Assert.assertNotEquals(0,cartRepository.findById(cart1.getOrder().getOrderId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartManyToOneUser(){
+        //Given
+        user1.setOrdersInsideTheCart(cart1);
+        user1.setOrdersInsideTheCart(cart2);
+        cart1.setUser(user1);
+        cart2.setUser(user1);
+        //When
+        userDao.save(user1);
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        //Then
+        Assert.assertEquals(2,user1.getOrdersInsideTheCart().size());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            userDao.deleteById(user1.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Test
+    public void testCartManyToManyProducts(){
+        //Given
+        product1.getListOfCarts().add(cart1);
+        product1.getListOfCarts().add(cart2);
+        product2.getListOfCarts().add(cart1);
+        product2.getListOfCarts().add(cart2);
+        cart1.getProductsList().add(product1);
+        cart1.getProductsList().add(product2);
+        cart2.getProductsList().add(product1);
+        cart2.getProductsList().add(product2);
+        //When
+        userDao.save(user1);
+        userDao.save(user2);
+        cartRepository.save(cart1);
+        Long product1Id = product1.getId();
+        cartRepository.save(cart2);
+        Long product2Id = product2.getId();
+        //Then
+        Assert.assertNotEquals(0.0,product1Id);
```

Comment:
0.0 to double, a potrzebujesz zero o typie Long. Trzeba wprowadzić zmienną Long zero = 0L; i nią się posługiwać w teście

Reply:
Long groupId = group1.getGroupId();
        Long productId1 = product1.getId();
        Long productId2 = product2.getId();

        //THEN
        Assert.assertNotEquals(0.0, groupId);
        Assert.assertNotEquals(0.0, productId1);
        Assert.assertNotEquals(0.0, productId2);

Ola takie same rozwiazanie zastosowalas w  w swoich testach encjiGroup, jest to wgrane do mastera, chyba ze to nie Ty robilas testy tej encji to przepraszam w kazdym badz razie jest to w masterze